core/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -141,4 +142,49 @@
      */
     boolean readOnly();
 
+    /**
+     * Creates a snapshot of the shard based on the index commit point.
+     * <p>
+     * The index commit point can be obtained by using {@link org.elasticsearch.index.engine.Engine#snapshotIndex} method.
+     * Repository implementations shouldn't release the snapshot index commit point. It is done by the method caller.
+     * <p>
+     * As snapshot process progresses, implementation of this method should update {@link IndexShardSnapshotStatus} object and check
+     * {@link IndexShardSnapshotStatus#aborted()} to see if the snapshot process should be aborted.
+     *
+     * @param shard               shard to be snapshotted
+     * @param snapshotId          snapshot id
+     * @param snapshotIndexCommit commit point
+     * @param snapshotStatus      snapshot status
+     */
+    void snapshot(IndexShard shard, SnapshotId snapshotId, IndexCommit snapshotIndexCommit, IndexShardSnapshotStatus snapshotStatus);
```

Comment:
Maybe just rename the method to `snapshotShard` or something.  I know it takes the IndexShard, that that really helps.